### PR TITLE
Add the consolidated release notes for 2.4.0 release

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add the consolidated release notes for 2.4.0 release

### Issues Resolved
Part of #2649 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+
+
+### OpenSearch Security Dashboards Plugin
+* Initial commit for multiple authentication ([#1110](https://github.com/opensearch-project/security-dashboards-plugin/pull/1110))
+* Saved Object Aggregation View ([#1146](https://github.com/opensearch-project/security-dashboards-plugin/pull/1146))
+* [Saved Object Aggregation View] Use namespace registry to add tenant filter ([#1169](https://github.com/opensearch-project/security-dashboards-plugin/pull/1169))
+* Move tenant-related utils to common folder ([#1184](https://github.com/opensearch-project/security-dashboards-plugin/pull/1184))
+
+
+### OpenSearch SQL
```
Comment (zelinh) [inline release-notes/opensearch-release-notes-2.4.0.md:92]: Need confirmation from SQL team regarding of the features part. @dai-chen

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+* feature: Add form generator & advanced settings to speed up development in common form case ([#329](https://github.com/opensearch-project/index-management-dashboards-plugin/pull/329))
+
+
+### OpenSearch k-NN
+* Merge efficient filtering from feature branch ([#588](https://github.com/opensearch-project/k-NN/pull/588))
+* Add groupId to pluginzip publication ([#578](https://github.com/opensearch-project/k-NN/pull/578))
+* Added sample perf-test configs for faiss-ivf, faiss-ivfpq, lucene-hnsw ([#555](https://github.com/opensearch-project/k-NN/pull/555))
+* Adding OSB index specification json for lucene hnsw ([#552](https://github.com/opensearch-project/k-NN/pull/552))
+* Adding k-NN engine stat ([#523](https://github.com/opensearch-project/k-NN/pull/523))
+
```
Comment (alicejw-aws) [inline release-notes/opensearch-release-notes-2.4.0.md:188]:
```suggestion
* Add filtering in Lucene HNSW k-NN ([#519](https://github.com/opensearch-project/k-NN/issues/519)
```

Comment (zelinh) [reply release-notes/opensearch-release-notes-2.4.0.md:188]: @alicejw-aws This looks like an issue but not a PR commit. @martin-gaievski Could you please help us confirm we do have such change in 2.4.0. Thanks.

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+## Release Highlights
+
+OpenSearch 2.4.0 is the first release with OpenSearch Security Analytics & Security Analytics Dashboards.
+Security Analytics consist of two plugins, security-analytics backend plugin for OpenSearch, and a securityAnalyticsDashboards frontend plugin for OpenSearch Dashboards.
+The Security Analytics Dashboards plugin lets you manage your Security Analytics plugin to generate critical security insights from their existing security event logs (such as firewall logs, windows logs, authentication audit logs, etc.) directly from OpenSearch Dashboards.
+
+## Release Details
+
+[OpenSearch and OpenSearch Dashboards 2.4.0](https://opensearch.org/versions/opensearch-2-4-0.html) includes the following features, experimental features, enhancements, bug fixes, infrastructure, documentation, maintenance, and refactoring updates.
```
Comment (zelinh) [inline release-notes/opensearch-release-notes-2.4.0.md:11]: This hyperlink to direct users to the artifacts download page is suggested in https://github.com/opensearch-project/opensearch-build/issues/2707

File: release-notes/opensearch-release-notes-2.4.0.md (hunk 1 above)
Comment (dai-chen) [reply release-notes/opensearch-release-notes-2.4.0.md:92]: @zelinh I confirmed with Ani that all items in https://github.com/opensearch-project/sql/blob/2.4/release-notes/opensearch-sql.release-notes-2.4.0.0.md#features are not experimental and should be present in `features` section here. If you want to keep one line for each, you can use the subsection title as feature name. Thanks!

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 4]
```diff
@@ -0,0 +1,704 @@
+# OpenSearch and Dashboards 2.4.0 Release Notes
+
+## Release Highlights
+
```
[the remark this answers is not in the document]
Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.4.0.md:4]: plugins, a security-analytics

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+The OpenSearch 2.4.0 release adds a number of new capabilities to help you enhance your search, analytics, and observability solutions. This release also marks the debut of the software’s first distribution for Windows environments, first security analytics plugin, first semantic search tools, and first solution for ingesting metric data from logs, plus new functionality to enhance cluster resiliency and protect and restore your data. Following are some highlights for this release. For a full list of the new and enhanced features, please review the _release notes_ [LINK TO BE ADDED].
```
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.4.0.md]: Change "For a full list of the new and enhanced features, please see below" or just remove that sentence and the previous sentence entirely.

File: release-notes/opensearch-release-notes-2.4.0.md (hunk 2 above)
Comment (heemin32) [reply release-notes/opensearch-release-notes-2.4.0.md:209]: We do have it in 2.4.0 in my understanding. It is already mentioned in release note line#18.
```
You can now create custom filters to refine your k-NN searches. The filtering is tightly integrated with the Lucene HNSW algorithm and allows you to apply k-NN searches more efficiently with better search results and performance.
```

File: release-notes/opensearch-release-notes-2.4.0.md (hunk 5 above)
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.4.0.md]: Removed.

File: release-notes/opensearch-release-notes-2.4.0.md [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+
+* Semantic search tools let you use machine learning (ML) to enhance document search results. The release adds functionality that enables OpenSearch to learn an ML model tuned to your documents. Using a pretrained query generator, you can create query-response pairs tailored to your documents, which an ML model can use to learn to order search results based on your data.
+* A new search relevance plugin lets you compare different search ranking techniques (including semantic search) side by side in the UI, so you can identify opportunities to fine-tune your results.
+* A new model serving framework allows you to upload your own ML models to OpenSearch.
+* OpenSearch Dashboards now supports multiple OpenSearch endpoints as data sources, allowing you to visualize data across different clusters in an environment through a single pane of glass.
+* A new security analytics plugin for OpenSearch and OpenSearch Dashboards provides tools to help you analyze security event logs, identify attack signatures, visualize threat patterns, and create security alerts.
```
[the remark this answers is not in the document]
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.4.0.md]: Updated. Thanks.

File: release-notes/opensearch-release-notes-2.4.0.md (hunk 2 above)
Comment (martin-gaievski) [reply release-notes/opensearch-release-notes-2.4.0.md:209]: Yes, that's correct, filtering is part of the k-NN 2.4.0

**Pull request merged**